flink-runtime/src/main/java/org/apache/flink/runtime/state/metainfo/StateMetaInfoSnapshotReadersWriters.java

```diff
@@ -139,10 +132,10 @@ private static StateMetaInfoReader getLegacyKeyedStateMetaInfoReader(int readVer
     private static StateMetaInfoReader getLegacyOperatorStateMetaInfoReader(int readVersion) {
```

Comment:
may as well drop this method, no? It _always_ throws an exception.

Reply:
I've simplified the code even a bit further (please take a look at the fixup commit)